spec/requests/vendor_api/reject_application_spec.rb

```diff
@@ -25,8 +25,17 @@
     end
   end
 
+  it 'returns an error when trying to transition to an invalid state' do
+    application_choice = create(:application_choice, status: 'rejected')
+
+    post "/api/v1/applications/#{application_choice.id}/reject", params: {}
+
+    expect(response).to have_http_status(422)
+    expect(parsed_response).to be_valid_against_openapi_schema('BadRequestBodyResponse')
+  end
+
   it 'returns not found error when the application was not found' do
-    post '/api/v1/applications/non-existent-id/confirm-enrolment'
+    post '/api/v1/applications/non-existent-id/reject'
```

Comment:
Nice catch! 👍

Reply:
Just a lil fyi incase, Duncan has created a PR to fix this, see https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/288.